### PR TITLE
Direct SVG rendering

### LINKED ISSUE #11618: Directly render SVG paths
When clicking through pages, I observe SVGs flashing because apparently the icon rendering needs to wait on the ajax request done in JS to succeed. To prevent this, we should directly render the `<path>`s without `<use>` both on server and client which does require a bit of parsing of the generated spritemap but should otherwise be possible.

### PR DESCRIPTION
**Pull request by silverwind**
This solves the issue of flashing SVG icons on page (re)load seen in some browsers (Firefox at least):

Introduce `make svg` which calls a node script that compiles svg files to `public/img/svg` which are then discovered by the backend during startup. These files are vendored to not create a dependency on Node for the backend build.

On the frontend side, configure webpack using `raw-loader` so SVGs can be imported as string. Also moved our existing SVGs to `web_src/svg` for consistency.

Fixes: https://github.com/go-gitea/gitea/issues/11618

### CODE REVIEW

**Comment by silverwind** [comment]
I think I will use a small script ran on `make svg` to generate the svg icon build and then vendor the resulting files in `public/img/svg`. If we don't vendor, there would be a dependency on node/npm to build the backend which I want to avoid.

**Comment by silverwind** [comment]
Ready for review (assuming ci passes now).

**Comment by silverwind** [comment]
Added gif to OP.

**Review by lunny — APPROVED** [review]
Next PR we could allow custom svg icons.

**Comment by lunny** [comment]
Please resolve the conflicts

**Comment by silverwind** [comment]
> Next PR we could allow custom svg icons.

That's already possible and in use for the custom lock icons we use. Currently they are in `assets/svg` and will be in `web_src/svg` after this PR.

File: package.json [hunk 1]
```diff
@@ -56,7 +54,8 @@
     "eslint-plugin-unicorn": "20.1.0",
     "stylelint": "13.6.1",
     "stylelint-config-standard": "20.0.0",
-    "updates": "10.2.15"
+    "updates": "10.2.15",
+    "svgo": "1.3.2"
```
Comment (silverwind) [inline package.json]:
```suggestion
    "svgo": "1.3.2",
    "updates": "10.2.15"
```

File: modules/svg/svg.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package svg
+
+// SVGs contains discovered SVGs
+var SVGs map[string]string
+
+// Init discovers SVGs
```
Comment (silverwind) [inline modules/svg/svg.go]:
```suggestion
// Init discovers SVGs and populates the `SVGs` variable
```

**Comment by silverwind** [comment]
Rebased

**Review by lafriks — APPROVED** [review]

**Pull request merged**